added chat component

xinkin opened the pull request:
I created a branch from my forked repo

vrajdesai78 commented:
messages are rendering twice, can you fix it. Also, I would suggest you improve the UI like how we have on our prod app. 

![Screenshot 2023-10-18 at 3 02 22 PM](https://github.com/Huddle01/Audio-spaces-example-app/assets/43074241/46cc6729-12be-411e-b4ec-5a9c15736969)
![Screenshot 2023-10-18 at 3 02 34 PM](https://github.com/Huddle01/Audio-spaces-example-app/assets/43074241/0f2341b4-437b-41fa-9af2-37618858c377)

xinkin commented:
I have made the necessary changes: fixed username rendering twice, added SVG, shortened the width, made UI palette similar to prod, please see to it @vrajdesai78

xinkin commented:
![Screenshot 2023-10-21 at 4 09 24 AM](https://github.com/Huddle01/Audio-spaces-example-app/assets/91452674/b807f3c1-6cff-4cd9-af95-32e155fb32a7)
![Screenshot 2023-10-21 at 4 09 10 AM](https://github.com/Huddle01/Audio-spaces-example-app/assets/91452674/370705e3-7423-4c35-bae8-85c4018eafff)
![Screenshot 2023-10-21 at 4 08 54 AM](https://github.com/Huddle01/Audio-spaces-example-app/assets/91452674/fd3f6eb7-a731-4b94-b072-a07f6cd04282)
here are a few screenshots of the same.

vrajdesai78 commented:
> ![Screenshot 2023-10-21 at 4 09 24 AM](https://user-images.githubusercontent.com/91452674/277063741-b807f3c1-6cff-4cd9-af95-32e155fb32a7.png) ![Screenshot 2023-10-21 at 4 09 10 AM](https://user-images.githubusercontent.com/91452674/277063751-370705e3-7423-4c35-bae8-85c4018eafff.png) ![Screenshot 2023-10-21 at 4 08 54 AM](https://user-images.githubusercontent.com/91452674/277063754-fd3f6eb7-a731-4b94-b072-a07f6cd04282.png) here are a few screenshots of the same.

UI looks better now, still you need to hide it on clicking a chat button in bottom right, like how we do for peers list. Still if I send a message it is rendering twice in the first peer. 

![image](https://github.com/Huddle01/Audio-spaces-example-app/assets/43074241/890f5e5f-6e24-4cdf-8e01-47f63c70678a)

xinkin commented:
@vrajdesai78 I am adding the show and hide feature but regards the twice message rendering I have checked my states thoroughly and I fixed the rendering of empty strings, I am not sure why you are having the message been rendered twice, its not an issue in my machine

vrajdesai78 commented:
@xinkin just a minor change

Instead of reading `data:received` event in chat.tsx and adding chats, use that code inside `[roomId]/page.tsx` as shown below. Remove the event listener from `chat.tsx`.  

![image](https://github.com/Huddle01/Audio-spaces-example-app/assets/43074241/4c28834f-c6c3-4014-a373-191f03e49963)

vrajdesai78 commented on src/components/BottomBar/BottomBar.tsx:
```diff
@@ -49,36 +53,38 @@ const BottomBar: React.FC<BottomBarProps> = () => {
+      </div>
 
       {/* Bottom Bar Center */}
       <div className="flex items-center mr-20 gap-4">
```
Remove `mr-20`

xinkin commented:
@vrajdesai78 made the changes please review

vrajdesai78 commented on src/components/Chat/Chat.tsx:
```diff
@@ -46,27 +43,16 @@ const Chat = () => {
-    }
-  });
-
   console.log(chatMessages);
```
remove this unnecessary console logs

xinkin commented:
@vrajdesai78 removed unnecessary console logs

vrajdesai78 commented:
Great work @xinkin, LGTM 🚀
